src/memote/metabolites.jl

```diff
@@ -0,0 +1,120 @@
+#=
+This file contains a collection of tests based on Memote. See Lieven, C., Beber,
+M.E., Olivier, B.G. et al. MEMOTE for standardized genome-scale metabolic model
+testing. Nat Biotechnol 38, 272–276 (2020).
+https://doi.org/10.1038/s41587-020-0446-y for details.
+=#
+
+"""
+$(TYPEDSIGNATURES)
+
+Return a list of all boundary reactions that allow flux into the model and
+create a metabolite. Assume that boundary reactions only create a single
+metabolite. Set the kwarg `only_into=false` to also return metabolites that can
+be produced by the model under default conditions.
+"""
+function metabolites_medium_components(model; only_into=true)
+    mets = String[]
+    for (rid, lb, ub) in zip(reactions(model), bounds(model)...)
+        if is_boundary(model, rid)
+            mid = first(keys(reaction_stoichiometry(model, rid))) 
+            only_into && lb < 0 && ub <= 0 && push!(mets, mid)
+            !only_into && lb < 0 && push!(mets, mid)
+        end
+    end
```

Comment:
this can be taken as a generating cycle

```julia
[first(keys(reaction_stoichiometry(model, rid)))
  for (rid,lb,ub)=zip(reactions.........)
  if is_boundary(model,rid)
  if influx && lb<0 && ub<=0  || !influx && lb<0
]
```

Reply:
too complicated